cmake/modules/AddSwift.cmake

```diff
@@ -1039,14 +1034,23 @@ function(_add_swift_library_single target name)
     set_target_properties("${target}"
       PROPERTIES
       INSTALL_NAME_DIR "${install_name_dir}")
-  elseif("${SWIFTLIB_SINGLE_SDK}" STREQUAL "LINUX" AND NOT "${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
+  elseif("${SWIFTLIB_SINGLE_SDK}" STREQUAL "LINUX")
     set_target_properties("${target}"
       PROPERTIES
       INSTALL_RPATH "$ORIGIN:/usr/lib/swift/linux")
   elseif("${SWIFTLIB_SINGLE_SDK}" STREQUAL "CYGWIN")
     set_target_properties("${target}"
       PROPERTIES
       INSTALL_RPATH "$ORIGIN:/usr/lib/swift/cygwin")
+  elseif("${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
+    # Only set the install RPATH if cross-compiling the host tools, in which
+    # case both the NDK and Sysroot paths must be set.
+    if(NOT "${SWIFT_ANDROID_NDK_PATH}" STREQUAL "" AND
+       NOT "${SWIFT_ANDROID_NATIVE_SYSROOT}" STREQUAL "")
+      set_target_properties("${target}"
+        PROPERTIES
+        INSTALL_RPATH "$ORIGIN:${SWIFT_ANDROID_NATIVE_SYSROOT}/usr/lib")
+    endif()
```

Comment:
Two things:
- Please don’t apply RPATH when building with the NDK. Let's try not to change the behaviour when cross-compiling. Also, I don't think the RPATH for cross-compiling can refer to `${SWIFT_ANDROID_NATIVE_SYSROOT}`, because that variable will be empty, and `/usr/lib` doesn't exist in Android (also, the system library paths should not be added to the RPATH, because they are already searched anyway).
- I'm very bad with logic, but I don’t think this branch is going to be taken at all, writing it in a more common language, you are doing `if (swift_android_ndk_path != "" and swift_android_native_sysroot != "")`, that is, when both `SWIFT_ANDROID_NDK_PATH` and `SWIFT_ANDROID_NATIVE_SYSROOT` are set to something. I would recommend changing the check to only check for the native sysroot.

Reply:
See the comment I had written above in this file: this is only going to kick in when the Android NDK is being used to cross-compile the host tools, ie the Swift compiler itself, to run on an Android host, ie Termux. As such, it shouldn't affect you when cross-compiling the Swift runtime/stdlib nor will it add the Android system library paths you mention.

If it wasn't clear already, here's the full Boolean truth table for how SWIFT_ANDROID_NDK_PATH and SWIFT_ANDROID_NATIVE_SYSROOT are used throughout this pull:
Both unset - Android is not being used
only NDK path set - cross-compiling the runtime/stdlib alone
only Sysroot path set - natively compiling the compiler _and_ runtime/stdlib on an Android host, ie in the Termux app like I've been doing
both NDK and Sysroot paths set - cross-compiling both the host tools and runtime/stdlib for an Android host

I only want this install rpath set for that last case, which is how the official Termux packages are built, which is why I check that both are set. Since you only care about the second case, where you don't set the new Android Sysroot path I added for Android host tools, this shouldn't affect you at all.

Reply:
Can you provide a link to some document about how Termux packages are supposed to be built? I find some minimal documentation in the wiki, but nothing related to rpath. If I understand correctly, they are trying to move away from `LD_LIBRARY_PATH` to use `RUNPATH`, which (IMO) is not the right thing to do, but if you can point me to any documentation, I will try to understand.

Reply:
The move to rpath is pretty new, I think [the best doc is the README link I gave you above, along with the github issue it links to](https://github.com/termux/termux-packages/blob/master/README.md#android-7-branch-is-in-alpha-testing-):
```
It is highly recommended to try to execute programs without
LD_LIBRARY_PATH set. Most of them should continue to run since
DT_RUNPATH field compiled-in ELF binary is used. Though, some
programs are not working due to missing DT_RUNPATH, please
check list in the related issue
```
That rpath github issue was just closed yesterday, saying that it's all working now.

That said, I'm mainly adding this rpath because that's what this project does for other OS's, as Saleem noted above, though of course Termux is moving to rpath also. As such, this INSTALL_RPATH patch is done speculatively, as I have not cross-compiled the Swift host tools yet.